src/Compiler/Utilities/range.fs

```diff
@@ -193,28 +193,29 @@ type FileIndexTable() =
     //
     // TO move forward we should eventually introduce a new type NormalizedFileName that tracks this invariant.
     member t.FileToIndex normalize filePath =
-        match fileToIndexTable.TryGetValue filePath with
-        | true, idx -> idx
-        | _ ->
-
-            // Try again looking for a normalized entry.
-            let normalizedFilePath =
-                if normalize then
-                    FileSystem.NormalizePathShim filePath
-                else
-                    filePath
+        lock fileToIndexTable
```

Comment:
It looks like you submitted a significantly better version of this fix that locks only when the filepath is not found on first lookup.  I think there may be a still better solution, because it seems that the indexToFileTable is where the underlying race occurred, and that would grab a lock only when a completely new normalized path is found I think.